include/SH3/system/glprogram.hpp

```diff
@@ -73,6 +73,14 @@ namespace sh3_gl
          */
         void Unbind();
 
+        /**
+         *  Gets the OpenGL generated Program ID from instance of this (shader) class so that
+         *  we can send/get data from outside of it.
+         *
+         *  return @ref programID The ID of this shader program.
+         */
+        inline GLuint GetProgramID(){return programID;}
```

Comment:
`inline` gets implied, so we don't need to specify that.
This function can be `const` though, which cannot be inferred by the compiler.

Reply:
Oh I see, so the compiler is smart enough to know that it should be inlined?? I've been reading a bit of old C++ lately so excuse the verbose syntax.

Reply:
Methods that are defined in the class definition are treated as specified with inline.

But the inline specifier is mostly just required to specify appropriate linkage. Compilers nowadays use heuristics based on the body and caller locations to determine when to inline functions. Having the definition available when it's called is essential for inlining though, unless you're compiling with LTO.